cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp

```diff
@@ -42,51 +42,46 @@ struct infer_ops {
     using descriptor_base_t = descriptor_base<task_t>;
 
     void check_preconditions(const Descriptor& params, const input_t& input) const {
-        if (!(input.get_data().has_data())) {
-            throw domain_error("Input data should not be empty");
-        }
-        if (input.get_model().get_support_vector_count() < 0) {
-            throw invalid_argument("Input model support_vector_count should be >= 0");
+        using msg = dal::detail::error_messages;
+
+        // TODO: Should be invariant of model
```

Comment:
do you mean that this should be checked in the model and in the kernel functions?